nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/RunNiFi.java

```diff
@@ -1027,19 +1028,24 @@ public boolean accept(final File dir, final String filename) {
         cmd.add("-Dorg.apache.nifi.bootstrap.config.log.dir=" + nifiLogDir);
         cmd.add("org.apache.nifi.NiFi");
         if (props.containsKey(NIFI_BOOTSTRAP_SENSITIVE_KEY) && !StringUtils.isBlank(props.get(NIFI_BOOTSTRAP_SENSITIVE_KEY))) {
-            cmd.add("-k " + props.get(NIFI_BOOTSTRAP_SENSITIVE_KEY));
+            File password_file = new File(confDir, "sensitive.key");
+
+            // Restrict permissions:
+            password_file.setReadable(false,false);
```

Comment:
Also: one could get a file descriptor on the `sensitive.key` file in the configuration directory before it's locked down, permissions-wise (think race condition and `inotify`). It needs to be created with initial permissions not allowing this---like in [the `java.nio.Files`-based approach example you posted](http://stackoverflow.com/a/32165319/70465).